core/services/keeper/registry_interface.go

```diff
@@ -119,8 +125,39 @@ func (rw *RegistryWrapper) GetActiveUpkeepIDs(opts *bind.CallOpts) ([]*big.Int,
 		}
 		return activeUpkeeps, nil
 	case RegistryVersion_1_2:
-		// TODO (sc-37024): Get active upkeep IDs from contract in batches
-		return rw.contract1_2.GetActiveUpkeepIDs(opts, big.NewInt(0), big.NewInt(0))
+		// fetch the current block number so batched GetActiveUpkeepIDs calls can be performed on the same block
+		header, err := rw.evmClient.HeaderByNumber(context.Background(), nil)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to fetch EVM block header")
+		}
+		if opts != nil {
+			opts.BlockNumber = header.Number
```

Comment:
Small nitpick, but it looks like we are always overriding the block number in `opts`. We should only overwrite the block number if it doesn't exist. That will allow someone to explicitly set the block height when making this call.

Reply:
also, we only need to fetch the block number if one was not provided in `opts`

Reply:
make sense